core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -568,6 +571,7 @@ class LogManager(logDirs: Array[File],
           log
         } catch {
           case e: IOException =>
+            error(s"Error while creating log for $topicPartition in dir ${dataDir.getAbsolutePath}", e)
```

Comment:
You said that these exceptions would previously be logged only if they are caught. Can we not fix this in one place instead of multiple places? The former would be more robust.